iocage_lib/ioc_common.py

```diff
@@ -107,6 +108,78 @@ def logit(content, _callback=None, silent=False, exception=RuntimeError):
     _callback(content, exception)
 
 
+def clone_repo(ref, repo_url, destination, depth=None, callback=None):
```

Comment:
I'm not so sure this belongs in common, after all, only the IOCPlugin module will ever be doing this.

Reply:
I can have this as a static method in the plugin interface, please advise what you think is better in this case. But it would seem weird calling `IOCPlugin._clone_repo`.

Reply:
Well perhaps a different method name is appropriate, but it really should live in that module. It’s a private method for just it

Reply:
Well this would be the exact invocation, `IOCPlugin._clone_repo(branch, repo_url, destination, depth, callback)` with params. Should i proceed ?

Reply:
It's fine to me, @william-gr do you have a better suggestion for a method name?

Reply:
Looks ok to me